docs/quick-start.md

```diff
@@ -0,0 +1,189 @@
+if [[ ! -f "${DIR}/metadata" ]]; then
+touch "${DIR}/metadata"
+truncate -s 2G "${DIR}/metadata"
+fi
+
+# Find/associate a loop device with our data volume.
+DATADEV="$(sudo losetup --output NAME --noheadings --associated ${DIR}/data)"
+if [[ -z "${DATADEV}" ]]; then
+    DATADEV="$(sudo losetup --find --show ${DIR}/data)"
+fi
+
+# Find/associate a loop device with our metadata volume.
+METADEV="$(sudo losetup --output NAME --noheadings --associated ${DIR}/metadata)"
+if [[ -z "${METADEV}" ]]; then
+    METADEV="$(sudo losetup --find --show ${DIR}/metadata)"
+fi
+
+# Magic calculations, for more information go and read
+# https://www.kernel.org/doc/Documentation/device-mapper/thin-provisioning.txt
+SECTORSIZE=512
+DATASIZE="$(blockdev --getsize64 -q ${DATADEV})"
+LENGTH_SECTORS=$(bc <<< "${DATASIZE}/${SECTORSIZE}")
+DATA_BLOCK_SIZE=128
+# picked arbitrarily
+# If free space on the data device drops below this level then a dm event will
+# be triggered which a userspace daemon should catch allowing it to extend the
+# pool device.
+LOW_WATER_MARK=32768
+
+THINP_TABLE="0 ${LENGTH_SECTORS} thin-pool ${METADEV} ${DATADEV} ${DATA_BLOCK_SIZE} ${LOW_WATER_MARK} 1 skip_block_zeroing"
+echo "${THINP_TABLE}"
+
+# If thinpool does not exist yet, create one.
+if ! $(dmsetup reload "${POOL}" --table "${THINP_TABLE}"); then
+    sudo dmsetup create "${POOL}" --table "${THINP_TABLE}"
+fi
+```
+
+I hope all my comments are enough to understand what this script does.
+
+### Configuration
+
+```toml
+# /etc/containerd/config-dev.toml
+version = 2
+
+root = "/var/lib/containerd-dev"
+state = "/run/containerd-dev"
+
+[grpc]
+  address = "/run/containerd-dev/containerd.sock"
+
+[metrics]
+  address = "127.0.0.1:1338"
+
+[plugins]
+  [plugins."io.containerd.grpc.v1.cri".containerd]
+    snapshotter = "devmapper"
+  [plugins."io.containerd.snapshotter.v1.devmapper"]
+    pool_name = "dev-thinpool"
+    base_image_size = "10GB"
+    root_path = "/var/lib/containerd-dev/snapshotter/devmapper"
+
+[debug]
+  level = "debug"
+```
+### Start containerd
+
+```
+# Just to make sure all the directories are there.
+sudo mkdir -p /var/lib/containerd-dev/snapshotter/devmapper
+sudo mkdir -p /run/containerd-dev/
+
+sudo containerd --config /etc/containerd/config-dev.toml
+```
+
+To reach our new dev containerd, we have to specify the `--address` flag,
+for example:
+
+```
+sudo ctr \
+    --address=/run/containerd-dev/containerd.sock \
+    --namespace=reignite \
+    content ls
+```
+
+To make it easier, here is an alias:
+
+```
+alias ctr-dev="sudo ctr --address=/run/containerd-dev/containerd.sock"
+```
+
+## Start Firecracker
+
+```
+rm /tmp/firecracker.socket && \
+  firecracker --api-sock /tmp/firecracker.socket
+```
+
+## Set up and start reignite
+
+```
+go mod download
+make build
+
+./bin/reignited run \
+  --containerd-socket=/run/containerd-dev/containerd.sock \
+  --firecracker-api=/tmp/firecracker.socket \
+  --parent-iface=tap0
```

Comment:
If using macvtap (recommended) then this will need to be the interface for that you use for your network connection.